chart/kubeapps/templates/kubeapps-jobs-upgrade.yaml

```diff
@@ -0,0 +1,52 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "kubeapps.kubeapps-jobs-upgrade.fullname" . }}
+  annotations:
+    helm.sh/hook: post-upgrade
+    helm.sh/hook-delete-policy: hook-succeeded
+  labels:
+    app: {{ template "kubeapps.kubeapps-jobs-upgrade.fullname" . }}
+    chart: {{ template "kubeapps.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  template:
+    metadata:
+      labels:
+        app: {{ template "kubeapps.kubeapps-jobs-upgrade.fullname" . }}
+        release: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: kubectl
+        image: {{ template "kubeapps.image" .Values.hooks.image }}
+        command:
+        - /bin/sh
+        - -c
+        args:
+        - |-
+            # Recreate MongoDB secret if doesn't exists (happens for chart versions < 0.5.2)
+            if kubectl get secrets -n {{ .Release.Namespace }} {{ .Values.mongodb.existingSecret }}; then
+              # Secret exists, do nothing
+              echo "MongoDB secret found"
+            else
+              # Secret doesn't exists, create it and restart mongodb
+              kubectl create secret generic -n {{ .Release.Namespace }} {{ .Values.mongodb.existingSecret }} --from-literal mongodb-root-password=`cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1`
+              kubectl delete pods -n {{ .Release.Namespace }} -l app=mongodb,release={{ .Release.Name }}
+              # Re-sync repositories reseting resyncRequests
+              kubectl get apprepositories -o=name -n kubeapps {{ .Release.Namespace }} | xargs kubectl patch $1 -n {{ .Release.Namespace }} --type merge -p '{ "spec": { "resyncRequests": 0 } }'
```

Comment:
Does the controller retry and so on if mongodb is down? Note that a new mongodb image will need to be downloaded so from `k delete pod mongodb` to being ready it might take some time. Time that the sync could fail.

Reply:
I added a check for the rollout status